cardano-testnet/src/Testnet/Babbage.hs

```diff
@@ -0,0 +1,386 @@
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE TypeApplications #-}
+
+{-# OPTIONS_GHC -Wno-unused-local-binds -Wno-unused-matches #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Testnet.Babbage
+  ( TestnetOptions(..)
+  , defaultTestnetOptions
+  , TestnetNodeOptions(..)
+  , defaultTestnetNodeOptions
+
+  , TestnetRuntime (..)
+  , TestnetNode (..)
+  , Wallet(..)
+
+  , testnet
+  ) where
+
+import           Control.Lens ((.~))
+import           Control.Monad (Monad (..), forM_, void, forM, (=<<), when, fmap, return)
+import           Data.Aeson ((.=))
+import           Data.Bool (Bool(..))
+import           Data.Eq (Eq)
+import           Data.Function (($), (.))
+import           Data.Functor ((<&>))
+import           Data.Int (Int)
+import           Data.List (unzip5, zipWith6)
+import           Data.Maybe (Maybe(Just))
+import           Data.Ord (Ord((<=)))
+import           Data.Semigroup (Semigroup((<>)))
+import           Data.String (String)
+import           GHC.Float (Double)
+import           Hedgehog.Extras.Stock.IO.Network.Sprocket (Sprocket (..))
+import           Hedgehog.Extras.Stock.Time (showUTCTimeSeconds)
+import           System.FilePath.Posix ((</>))
+import           System.IO (FilePath)
+import           Text.Show (Show(show))
+
+import qualified Data.Aeson as J
+import qualified Data.HashMap.Lazy as HM
+import qualified Data.Aeson.Lens as J
+import qualified Data.List as L
+import qualified Data.Time.Clock as DTC
+import qualified Hedgehog as H
+import qualified Hedgehog.Extras.Stock.Aeson as J
+import qualified Hedgehog.Extras.Stock.IO.File as IO
+import qualified Hedgehog.Extras.Stock.IO.Network.Sprocket as IO
+import qualified Hedgehog.Extras.Stock.OS as OS
+import qualified Hedgehog.Extras.Stock.String as S
+import qualified Hedgehog.Extras.Test.Base as H
+import qualified Hedgehog.Extras.Test.File as H
+import qualified Hedgehog.Extras.Test.Process as H
+import qualified System.Info as OS
+import qualified System.IO as IO
+import qualified System.Process as IO
+import qualified Test.Process as H
+import qualified Testnet.Conf as H
+
+{- HLINT ignore "Reduce duplication" -}
+{- HLINT ignore "Redundant <&>" -}
+{- HLINT ignore "Redundant flip" -}
+{- HLINT ignore "Redundant id" -}
+{- HLINT ignore "Use let" -}
+
+
+data TestnetOptions = TestnetOptions deriving (Eq, Show)
+
+defaultTestnetOptions :: TestnetOptions
+defaultTestnetOptions = TestnetOptions
+
+data TestnetNodeOptions = TestnetNodeOptions deriving (Eq, Show)
+
+defaultTestnetNodeOptions :: TestnetNodeOptions
+defaultTestnetNodeOptions = TestnetNodeOptions
+
+data TestnetRuntime = TestnetRuntime
```

Comment:
Likewise for everything else that has been copy pasted.

Reply:
I left them as separate types initially because I was concerned there may be differences between the configuration data we use for different eras.

I'm more comfortable with making them the same type now.

Reply:
> may be differences between the configuration data we use for different eras.
We can encapsulate this with function like `createTestnetRuntime :: AnyCardanoEra -> TestnetRuntime`